docs/docs/apis/focusutils.md

```diff
@@ -0,0 +1,78 @@
+---
+id: apis/focusutils
+title: FocusUtils
+layout: docs
+category: Interfaces
+permalink: docs/apis/focusutils.html
+next: apis/input
+---
+
+This interface provides methods related to managing focusable components.
+
+## Methods
+``` javascript
+// Allows to register a component for being automatically focused during the
+// current mount cycle.
+// The same method which is used internally to handle autoFocus property for
+// the components which might support being focused (View, Button, etc.)
+
+// Moving the focus manually from the application code might lead to a complex
+// logic on the application level and to race conditions when several components
+// want to be focused on mount at the same time. Those race conditions might
+// result in an unreliable behaviour, especially for the screen reader users.
+// For example, when some element inside the dialog is focused, the screen reader
+// announces the dialog, but only when nothing inside the dialog was focused before,
+// and if the focus is moved inside the dialog, the second focused element
+// might interrupt the dialog announcement and the screen reader users won't be
+// able to understand where they are.
+// Another thing is that it is often required to have different autofocusing logic
+// on different platforms. For example it is often ok to automatically focus the
+// input field on desktop when something shows, but on mobile it will bring up the
+// keyboard which is not always the desired behaviour.
+
+// To tackle above we have this concept of smart automatic focus.
+// See the Types.AutoFocus enum definition:
+
+// export enum AutoFocus {
+//     No = 0,
+//     Yes = 1, // All platforms, any state of the keyboard navigation mode, default
+//              // priority, no delay.
+
+//     // We can target the keyboard navigation mode (if none of
+//     // WhenNavigatingWithKeyboard and WhenNavigatingWithoutKeyboard are specified,
+//     // both are assumed to be enabled).
+//     WhenNavigatingWithKeyboard,
+//     WhenNavigatingWithoutKeyboard,
+
+//     // We can target the platform (if none of the platforms are specified, all
+//     // platforms are assumed to be enabled).
+//     Android,
```

Comment:
Exposing specific platforms within an enumeration is not an approach we should be using. It's not scalable, and it goes against the principles of ReactXP. These should be removed. If the caller wants to make it conditional based on platform, they can include the check within their app code.